Search/SlidingWindow.js

```diff
@@ -0,0 +1,51 @@
+/**
+* Sliding Window:
+* This pattern involve creating a window which can either be
+* an array or numbers from one position to another.
+*
+* Depending on a certain condition, the window either increases
+* or closes (and a new window is created).
+*
+* Very useful for keeping track of a subset of data in an
+* array/string etc.
+*
+* Time Complexity: Best - O(n);
+*
+* Examples:
+* maxSubarraySum([1,2,5,2,8,1,5],2) // returns 10
+* maxSubarraySum([1,2,5,2,8,1,5],15) // returns null
+* maxSubarraySum([5,2,6,9],3) // returns 17
```

Comment:
This example won't work. Do you mean slidingWindow?

Reply:
Ohh... yes I did meant to write slidingWindow :)
Do you want me to rewrite it and send the pull request again?

Reply:
Just add more commits to this pr.

Reply:
I did the commit, hope it's all right now.